src/renderer/components/+catalog/catalog-entity.store.tsx

```diff
@@ -67,4 +67,35 @@ export class CatalogEntityStore extends ItemStore<CatalogEntityItem<CatalogEntit
 
     return this.loadItems(() => this.entities);
   }
+
+  /**
+   * Override ItemStore::loadItems, the only difference is that
+   * if isLoading is true, we await for it to be false and then continue
+   * loading.
+   * This avoids a bug where if there were two or more concurrent loadItems() calls,
+   * only the first one would complete.
+   */
+  protected async loadItems(
+    request: (() => Promise<CatalogEntityItem<CatalogEntity>[] | any>) | (() => CatalogEntityItem<CatalogEntity>[] | any),
```

Comment:
This could simplified to `() => (Promise<CatalogEntityItem<CatalogEntity>[]> | CatalogEntityItem<CatalogEntity>[])`

Reply:
Though I would say that it would be better to add a param to `super.loadItems` to bypass that `return`, I think that would be less fragile.

Reply:
Added parameter instead to `super.loadItems`